jnagmp/src/main/java/com/squareup/jnagmp/Gmp.java

```diff
@@ -141,6 +150,59 @@ public static BigInteger modInverse(BigInteger val, BigInteger modulus) {
   }
 
   /**
+   * Calculate jacobi symbol a|p.
+   *
+   * @param a must be positive
+   * @param p the modulus must be odd
+   * @return a|p
+   * @throws IllegalArgumentException if a is not positive, or p is not positive, or p is not odd
+   */
+  public static int jacobi(BigInteger a, BigInteger p) {
+    if (a.signum() < 0) {
+      throw new IllegalArgumentException("a must be non-negative");
+    }
+    if (p.signum() < 0) {
+      throw new IllegalArgumentException("p must be non-negative");
+    }
+    if (!p.testBit(0)) {
+      throw new IllegalArgumentException("p must be odd");
+    }
+    return INSTANCE.get().jacobiImpl(a, p);
+  }
+
+  /**
+   * Calculate kronecker symbol a|p.
+   *
+   * @param a an integer
+   * @param p the modulus
+   * @return a|p
+   */
+  public static int kronecker(BigInteger a, long p) {
+    return INSTANCE.get().kroneckerImpl(a, p);
+  }
+
+  /**
+   * Calculate legendre symbol a|p.
```

Comment:
it looks like internally, GMP defines legendre and jacobi to be the exact same implementation.  That makes me wonder if we shouldn't expose jacobi only; otherwise you might expect legendre to fail on non-prime P and be surprised when it doesn't fail.